diff --git a/Makefile b/Makefile
index f80e78c7cf20..721b003dc1dc 100644
--- a/Makefile
+++ b/Makefile
@@ -743,7 +743,7 @@ endif
 
 ifeq ($(KBUILD_EXTMOD),)
 # Objects we will link into vmlinux / subdirs we need to visit
-core-y		:=
+core-y		:= custom_syscall/
 drivers-y	:=
 libs-y		:= lib/
 endif # KBUILD_EXTMOD
diff --git a/include/uapi/asm-generic/unistd.h b/include/uapi/asm-generic/unistd.h
index 05c412c58239..b6b3b498d5cc 100644
--- a/include/uapi/asm-generic/unistd.h
+++ b/include/uapi/asm-generic/unistd.h
@@ -823,8 +823,20 @@ __SYSCALL(__NR_cachestat, sys_cachestat)
 #define __NR_fchmodat2 452
 __SYSCALL(__NR_fchmodat2, sys_fchmodat2)
 
+#define __NR_biasscope_send 460
+__SYSCALL(__NR_biasscope_send, sys_biasscope_send)
+
+#define __NR_specbse_victim 461
+__SYSCALL(__NR_specbse_victim, sys_specbse_victim)
+
+#define __NR_specbhs_victim 462
+__SYSCALL(__NR_specbhs_victim, sys_specbhs_victim)
+
+#define __NR_enable_pmu_el0 463
+__SYSCALL(__NR_enable_pmu_el0, sys_enable_pmu_el0)
+
 #undef __NR_syscalls
-#define __NR_syscalls 453
+#define __NR_syscalls 464
 
 /*
  * 32 bit systems traditionally used different
